python/lsst/ts/m2gui/controltab/tab_hardpoint_selection.py

```diff
@@ -0,0 +1,300 @@
+
+from lsst.ts.m2com import (
+    NUM_ACTUATOR,
+    NUM_HARDPOINTS_AXIAL,
+    NUM_TANGENT_LINK,
+    check_hardpoints,
+    read_yaml_file,
+    select_axial_hardpoints,
+)
+from PySide2.QtWidgets import QGroupBox, QVBoxLayout
+from qasync import asyncSlot
+
+from ..enums import LocalMode
+from ..model import Model
+from ..signals import SignalDetailedForce
+from ..utils import (
+    create_grid_layout_buttons,
+    create_group_box,
+    get_checked_buttons,
+    run_command,
+    set_button,
+)
+from .tab_default import TabDefault
+
+
+class TabHardpointSelection(TabDefault):
+    """Table of the hardpoint selection.
+
+    Parameters
+    ----------
+    title : `str`
+        Table's title.
+    model : `Model`
+        Model class.
+
+    Attributes
+    ----------
+    model : `Model`
+        Model class.
+    """
+
+    def __init__(self, title: str, model: Model) -> None:
+        super().__init__(title, model)
+
+        self._cell_geom: dict = dict()
+        self._hardpoints: list = list()
+
+        self._buttons_hardpoint_selection = [
+            set_button(name, None, is_checkable=True, is_adjust_size=True)
+            for name in self.model.get_actuator_default_status(False)
+        ]
+        self._buttons_hardpoint = {
+            "reset": set_button(
+                "Reset Default",
+                self._callback_reset_default,
+                tool_tip="Reset the default hardpoints",
+            ),
+            "apply": set_button(
+                "Apply Hardpoints",
+                self._callback_apply_hardpoints,
+                tool_tip="Apply the selected hardpoints",
+            ),
+            "suggest": set_button(
+                "Suggest Hardpoints",
+                self._callback_suggest_hardpoints,
+                tool_tip="Suggest the hardpoints",
+            ),
+        }
+
+        self.set_widget_and_layout()
+
+        self._set_signal_detailed_force(
+            self.model.utility_monitor.signal_detailed_force
+        )
+
+    @asyncSlot()
+    async def _callback_reset_default(self) -> None:
+        """Callback of the reset-default button to reset the selection of
+        hardpoints back to the default."""
+
+        await run_command(self._set_hardpoints, self._hardpoints)
+
+    def _set_hardpoints(self, hardpoints: list[int]) -> None:
+        """Set the hardpoints.
+
+        Parameters
+        ----------
+        hardpoints : `list`
+            List of the 0-based hardpoints. There are 6 actuators. The first
+            three are the axial actuators and the latter three are the tangent
+            links.
+        """
+
+        for idx, hardpoint in enumerate(self._buttons_hardpoint_selection):
+            hardpoint.setChecked(idx in hardpoints)
+
+    @asyncSlot()
+    async def _callback_apply_hardpoints(self) -> None:
+        """Callback of the apply-hardpoints button to apply the selected
+        hardpoints."""
+
+        await run_command(self._set_hardpoint_list)
```

Comment:
As run_command adds try/catch, you can either make from this @decorator..

@command
async def _call...

and then:

def command(funci, *args):
    try:
       func(*args)
    except Exception:
       xxx

same as the current run_command

Reply:
Thanks for the suggestion! Usually I try to avoid the use of decorator. Let me see it would be useful or not in a latter time. Thanks!